src/depackers/arc_unpack.c

```diff
@@ -129,14 +129,14 @@ static int arc_unpack_init(struct arc_data *arc, int init_width, int max_width,
     if(max_width < 9 || max_width > 16)
       return -1;
 
-    arc->tree = (struct arc_code *)calloc(1 << max_width, sizeof(struct arc_code));
+    arc->tree = (struct arc_code *)calloc((size_t)(1U << max_width), sizeof(struct arc_code));
     if(!arc->tree)
       return -1;
 
     for(i = 0; i < 256; i++)
     {
       struct arc_code *c = &(arc->tree[i]);
-      c->prev = (arc_uint16)ARC_NO_CODE;
+      c->prev = 0xffffUL;
       c->length = 1;
```

Comment:
I think this was just a placeholder value meant to be obvious during debug, as these codes *don't* have previous codes. Any length calculation will terminate before this value is ever used (`arc_unlzw_get_length`).

Replacing the constant with a magic number implies that value has a different purpose/intent than `ARC_NO_CODE`, so 0 or removing the line might be better (the tree is already initialized, `calloc`).

Reply:
OK, just removed the line now